Full documentation about the bridge. v2 preset api

Paty007gr opened the pull request:
Full revamp of the old documentation I wrote now with most if not all the components of presets

Paty007gr committed:
Update sidebars.js

Paty007gr committed:
Create presetsv2

Paty007gr committed:
Update and rename presetsv2 to presetsv2.md

Paty007gr committed:
Add FileMap to docs

Paty007gr committed:
Full explanation

Paty007gr committed:
Update presetsv2.md

Paty007gr committed:
Update presetsv2.md

Paty007gr committed:
Updated links

Paty007gr committed:
Small typo correction

Paty007gr committed:
Minor Changes...

Added category to the manifest format and changed the general paragraph

Paty007gr committed:
Revamp

Paty007gr committed:
Revamp

Paty007gr committed:
Addtions that werent added when revamp

Paty007gr committed:
Tab problem

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+| `description`                        | `String`              | Preset description                       |
+| `icon`                               | `String`              | Icon to show inside of the preset window |
+| `category`                           | `String`              | Category of the preset entity, item, ect |
+| `packTypes`                          | `String`	       | Which packs are needed for the preset    |
```
This should be an array

Paty007gr replied:
Which one the packTypes?

solvedDev replied:
Yep

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+- The Variables injected to the file
+
+The template file name is a string that will include the file name and the extension of the file name such as `.json` or `.lang` that will be created or expanded.
+The file path will define where the specified template file will be created. If the file is supposed to be expanded the path need to define what file its expanding for example `RP/texts/en_US.lang`.
```
Typo: "its" -> "it's"
"...for example..." -> (Example: ... )

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+The template file name is a string that will include the file name and the extension of the file name such as `.json` or `.lang` that will be created or expanded.
+The file path will define where the specified template file will be created. If the file is supposed to be expanded the path need to define what file its expanding for example `RP/texts/en_US.lang`.
+The variables that will be included in the specified template file will also need to be "Injected" so bridge. knows what variables need to be replaced with their corresponding values
```
"Injected" -> "injected"

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+### Fields
+
+Fields create input boxes in the preset window where the player can input some type of data like a image or text that will be assigned to the variable specified after the name of the input box these are the types of input:
```
"like a image" -> "like an image"

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+###### Text
+
+`textInput` is the default type of input is `text`. To create a `text` input type make a array with the name of the text box as a `string` followed by another `string` which is the variable where the user's input will be assigned to
```
First sentence is a bit messed up
"make a array" -> "make an array"
"is the variable where the user's input will be assigned to" -> "is the variable to which the user's input gets assigned"

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+###### File
+
+`fileInput` input types can be created by an array this will include the name of the input box, the input type. Last by creating a `object` properties are defined such as `accept`, `icon` these can only be used inside a `fileInput` input type.
```
This section is a bit messed up 🙃

Paty007gr replied:
What does the icon property do? (inside the fileInput)

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+###### Selection Box
+
+`selectInput` is an input type creates selection boxes or drop down lists that have a fixed amount of options. To make a selection box create a input box with a name and assign a variable to it, then create a properties object and define the type of input box as `selectInput` then a `default` value this is the default option the input box will be set. Create an array named `options` then for each ...
```
First sentence is messed up

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+###### Number Input
+
+`numberInput` is an input type that creates a slider this slider goes from a specified `min` number to a specified `max` number. This slider will count up or down by a specified `step` number. To create a `numberInput` create a new input box with the `numberInput` as the input `type` then put a `min`, `max`, `step` number.
```
First sentence is messed up

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+###### Switch
+
+`switch` is the simplest type of input which is a literal switch that can be turned on or off. The variable that is assigned to the input will get `0` or `1` based on the switch state (`0` for `off` and `1` for `on`). To create it set the input `type` to `switch`
```
Variable values are `true` or `false`, not `1`or `0`

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+
+### Icon
+Icons show up in the presets window. Presets use [Material Design Icons](https://materialdesignicons.com/) a collection of icons that can be defined in the manifest.json by typing `mdi-` followed by the icon's name.
```
"...use Material Desgin Icons, ..."

Paty007gr replied:
Why remove the link?

solvedDev replied:
Na, there's a comma missing afaik (I am not a native English speaker though). The link should definitely stay :)

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+### Variables
+
+Variables are able to be used in the `createFiles` and `expandFile` components of the `manifest.json they` can also be used any other files in the preset as long as they are "injected" in the file.
```
"`manifest.json they`" -> "`manifest.json`. They"
"... also be used any other files in the preset ..." -> "...also be used in any other files of the preset..."

solvedDev commented on content/docs/plugin-docs/json/presetsv2.md:
```diff
@@ -0,0 +1,228 @@
+
+
+
+Page written by Paty
```
The author tag on our website is working automatically because GitHub knows that you have written the file. You can delete this line :)

solvedDev reviewed (changes requested):
Thanks a lot for these thorough docs 👏

Paty007gr committed:
Update presetsv2.md

Paty007gr committed:
Small changes

Grammar corrections, Some rewriting, Added additionalModels to manifest components

Paty007gr committed:
Quick fields description change

Paty007gr committed:
Changed "player" to "user"

Paty007gr committed:
Added a better explanation of input properties

Paty007gr committed:
TAB PROBLEM AGAIN AAAAAAAAA

Paty007gr committed:
Update presetsv2.md
